docs/guide/cache/fastly.md

```diff
@@ -0,0 +1,464 @@
+cp generated_vcl_11_raw generated_vcl_11_json_only
+```
+
+Next, you need to edit generated_vcl_11_json_only in your favourite editor, remove anything before the json data and save.
+Then, do the same steps again for version no. 12 (or whatever version you want to diff version 11 against)
+
+Then replace \n in the files in order to give human-readable diffs:
+
+```
+cat generated_vcl_11_json_only |jq .content|perl -pe 's/\\n/\n/g' > generated_vcl_11_json_done
+cat generated_vcl_12_json_only |jq .content|perl -pe 's/\\n/\n/g' > generated_vcl_12_json_done
+```
+
+Finally, you can use GNU diff to and get a readable diff of the two versions:
+
+```
+diff -ruN generated_vcl_11_json_done generated_vcl_12_json_done
+```
+
+## Enabling Basic-Auth on Fastly.
+
+In order to enable basic-auth, you can use https://developer.fastly.com/solutions/examples/http-basic-auth as an example.
+
+Here is a step-by-step guide on how to make that work.
+
+Usernames and password can be stored inside vcl, but we'll instead store them in a [dictionary](https://docs.fastly.com/en/guides/working-with-dictionaries-using-the-web-interface#working-with-dictionaries-using-vcl-snippets)
+
+!!! note "In order to make this example work, you must run Ibexa DXP 3.3.16 or later, or 4.5."
+
+### Create the dictionary and activate the change:
+
+```
+fastly dictionary create --version=active --autoclone --name=basicauth
+fastly service-version activate --version=latest
+```
+
+From now on, our Fastly configuration has a dictionary named `basicauth`. The benefit of using a dictionary vs storing
+the usernames directly in vcl is that we can add/remove records to it without creating and activating new versions of our configuration.
+
+### Obtain the dictionary ID
+
+In order to add users to the dictionary, we first need to get the dictionary ID.
+
+```
+fastly dictionary list --version=active
+
+Service ID: KlUh0J1fnw1JY1aEQ0up
+Version: 3
+ID: ltC6Rg4pqw4qaNKF5tEW
+Name: basicauth
+Write Only: false
+Created (UTC): 2023-07-03 10:33
+Last edited (UTC): 2023-07-03 10:33
+```
+
+In the example above, the ID is `ltC6Rg4pqw4qaNKF5tEW`.
+
+
+### Create a record in the dictionary ( user name and password )
+
+```
+fastly dictionary-item create --dictionary-id=ltC6Rg4pqw4qaNKF5tEW --key=user1 --value=foobar1
+```
+
+### List dictionary records
+
+You may see the records of a dictionary using:
+
+```
+fastly dictionary-item list --dictionary-id=ltC6Rg4pqw4qaNKF5tEW33
+```
+
+Now we have a dictionary storing a user name and password. The next thing to do is to alter the Fastly .vcl configration.
+and add the basic-auth support. 
+In this example, we'll do that using snippets (https://docs.fastly.com/en/guides/about-vcl-snippets ) so that no changes
+are need in the .vcl files shipped by Ibexa DXP. You need two snippets, store these as files on your system:
```

Comment:
```suggestion
Now you have a dictionary storing a user name and password. The next thing to do is to alter the Fastly .vcl configration.
and add the basic-auth support. 
In this example, do that using snippets (https://docs.fastly.com/en/guides/about-vcl-snippets ) so that no changes
are need in the .vcl files shipped by Ibexa DXP. You need two snippets, store these as files on your system:
```